src/integrationtests/java/com/aws/greengrass/integrationtests/provisioning/ProvisionFromTestPlugin.java

```diff
@@ -0,0 +1,267 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@ExtendWith(GGExtension.class)
+public class ProvisionFromTestPlugin extends BaseITCase {
+
+    private Kernel kernel;
+
+    @TempDir
+    Path configurationFilePath;
+    @TempDir
+    Path certFilePath;
+    @TempDir
+    Path privateKeyPath;
+    @TempDir
+    Path rootCAPath;
+
+    @BeforeEach
+    void before() {
+        kernel = new Kernel();
+    }
+
+    @AfterEach
+    void after() {
+        if (kernel != null) {
+            // Executor service is not able to terminate threads in some tests within the default 30 seconds
+            kernel.shutdown(40);
+        }
+    }
+
+    @Order(1)
+    @Test
+    @SuppressWarnings("PMD.AvoidDeeplyNestedIfStmts")
+    void GIVEN_nucleus_started_with_testProvisioningPlugin_AND_without_config_params_THEN_device_runs_offline_mode() throws Throwable {
+        URL filepath = getClass().getResource("config_without_test_provisioning_plugin.yaml");
+        ConfigPlatformResolver.initKernelWithMultiPlatformConfig(kernel, filepath);
+        CountDownLatch logLatch =  new CountDownLatch(2);
+        CountDownLatch reverseLatch = new CountDownLatch(1);
+        DeviceConfiguration deviceConfiguration = kernel.getContext().get(DeviceConfiguration.class);
+        try (AutoCloseable listener = TestUtils.createCloseableLogListener((message) -> {
+            String messageString = message.getMessage();
+            if (messageString.contains("Device not configured to talk to AWS Iot cloud")) {
+                logLatch.countDown();
+            } else if ("Updated provisioning configuration".equals(messageString)) {
+                deviceConfiguration.onAnyChange((what, node) -> {
+                    if (node != null && WhatHappened.childChanged.equals(what)) {
+                        if (node.childOf(DEVICE_PARAM_THING_NAME) || node.childOf(DEVICE_PARAM_IOT_DATA_ENDPOINT)
+                                || node.childOf(DEVICE_PARAM_PRIVATE_KEY_PATH)
+                                || node.childOf(DEVICE_PARAM_CERTIFICATE_FILE_PATH) || node.childOf(DEVICE_PARAM_ROOT_CA_PATH)
+                                || node.childOf(DEVICE_PARAM_AWS_REGION)) {
+                            reverseLatch.countDown();
+                        }
+                    }
+                });
+                logLatch.countDown();
+            }
+        })) {
+
+            kernel.launch();
+            assertTrue(logLatch.await(2, TimeUnit.SECONDS));
+            assertFalse(reverseLatch.await(5, TimeUnit.SECONDS));
+        }
+    }
+
+    @Order(2)
+    @Test
+    void GIVEN_nucleus_started_with_testProvisioningPlugin_AND_plugin_throws_non_retryable_exception(ExtensionContext context) throws Throwable {
```

Comment:
Ni - could you make this into the GIVEN_WHEN_THEN format to make the expectation from this test clear?